### PR TITLE
[WIP] Add compare encoders example

### LINKED ISSUE #139: Comparison to factorization methods (e.g. SVD)
I used the notebook 02_fit_predict_plot_employee_salaries.html to compare the performance using SVD (with same number of components as MinHashEncoder). This could be added to the current or different notebook, if you agree. 

one-hot encoding
r2 score:  mean: 0.856; std: 0.034

target encoding
r2 score:  mean: 0.774; std: 0.033

similarity encoding
r2 score:  mean: 0.915; std: 0.012

minhash encoding
r2 score:  mean: 0.753; std: 0.025

svd encoding
r2 score:  mean: 0.856; std: 0.014

### PR DESCRIPTION
**Pull request by LilianBoulard**
Extracted from #546.

Fixes #139

### CODE REVIEW

File: examples/07_compare_encoders.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
+"""
+.. _example_compare_encoders
+
+============================================
+Understanding and comparing encoding methods
+============================================
+
+In this example, we will compare the mechanisms and use-cases of
+a classical encoding method, the |OneHotEncoder|,
```
Comment (GaelVaroquaux) [inline examples/07_compare_encoders.py:9]:
```suggestion
In this example, we compare various categorical encoders:
a classical encoding method, the |OneHotEncoder|,
```

File: examples/07_compare_encoders.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
+"""
+.. _example_compare_encoders
+
+============================================
+Understanding and comparing encoding methods
+============================================
```
Comment (GaelVaroquaux) [inline examples/07_compare_encoders.py:6]:
```suggestion
====================================================
Understanding and comparing categorical encoding methods
====================================================
```

File: examples/07_compare_encoders.py [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+In this example, we will compare the mechanisms and use-cases of
+a classical encoding method, the |OneHotEncoder|,
+and those implemented in *skrub*, that is, the |SimilarityEncoder|,
+the |GapEncoder| and the |MinHashEncoder|.
+
+.. note:
```
Comment (GaelVaroquaux) [inline examples/07_compare_encoders.py:13]: Maybe a seealso block rather than a note?

File: examples/07_compare_encoders.py [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+.. |MinHashEncoder| replace::
+    :class:`~skrub.MinHashEncoder`
+"""
+
+###############################################################################
+# Encoding dirty categorical variables
+# ------------------------------------
+#
+# As usual, let's first retrieve the dataset:
```
Comment (GaelVaroquaux) [inline examples/07_compare_encoders.py:39]:
```suggestion
# Let's first retrieve the dataset:
```

File: examples/07_compare_encoders.py [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+###############################################################################
+# Next up: import the |OneHotEncoder| and apply it to our sample:
+
+from sklearn.preprocessing import OneHotEncoder
+import pandas as pd
+
+one_hot = OneHotEncoder(sparse_output=False)
+sample_enc_ohe = one_hot.fit_transform(sample)
+
+# (make it look nice in Jupyter by wrapping the resulting array in a DataFrame)
```
Comment (GaelVaroquaux) [inline examples/07_compare_encoders.py:97]: I would rather we use the "set_output" method of scikit-learn transformer. Just call "one_hot.set_output(transform="pandas")" above

File: examples/07_compare_encoders.py [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+# The |SimilarityEncoder| is a generalization of the |OneHotEncoder|, which
+# uses a string similarity metric to encode the similarity between categories.
+# As we did for the one-hot, let's visualize what that means!
+
+from skrub import SimilarityEncoder
+
+sim_enc = SimilarityEncoder()
+sample_enc_sim = sim_enc.fit_transform(sample)
+
+# (make it look nice in Jupyter by wrapping the resulting array in a DataFrame)
```
Comment (GaelVaroquaux) [inline examples/07_compare_encoders.py:134]: Same thing. Let's use "set_output"

File: examples/07_compare_encoders.py [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+heatmap(
+    sample_enc_sim,
+    xticklabels=sim_enc.categories_[0],
+    yticklabels=sample["employee_position_title"],
+    ax=ax,
+)
+plt.tight_layout()
+
+###############################################################################
+# While this is a cool result, it suffers from the high computational cost
```
Comment (GaelVaroquaux) [inline examples/07_compare_encoders.py:165]: Let's be slightly less casual and avoid the word "cool" :)

**Review by GaelVaroquaux — COMMENTED** [review]
An early set of comments (I note that there still is the MinHashEcoder to implement, and also maybe a comparison of the performance of all encoders, as we have here https://dirty-cat.github.io/stable/auto_examples/01_dirty_categories.html#sphx-glr-auto-examples-01-dirty-categories-py )

Also, the documentation is not building. I am not sure why

**Comment by jovan-stojanovic** [comment]
> I can continue this work if that's ok, @jovan-stojanovic ?

Yes of course!

**Comment by jovan-stojanovic** [comment]
I will close this due to inactivity but also because the example comparing encoders is less important now that we have new features and that part of it was implemented in the first example. Thanks Lilian for your work!

**Pull request closed without merging**